packages/client/wallets/smart-wallet/src/blockchain/wallets/SendTransactionService.ts

```diff
@@ -0,0 +1,142 @@
+import { SmartAccountClient } from "permissionless";
+import { EntryPoint } from "permissionless/_types/types";
+import { SmartAccount } from "permissionless/accounts";
+import {
+    Abi,
+    Address,
+    BaseError,
+    Chain,
+    ContractFunctionRevertedError,
+    PublicClient,
+    TransactionReceipt,
+    Transport,
+} from "viem";
+
+import { CrossmintSDKError, WalletErrorCode } from "@crossmint/client-sdk-base";
+
+export type TransactionServiceTransactionRequest = {
+    address: Address;
+    abi: Abi;
+    functionName: string;
+    args: any;
+    value?: bigint;
+};
+
+/**
+ * Error thrown when a transaction fails to send.
+ * @param viemError The error thrown by the viem client.
+ */
+export class SendTransactionError extends CrossmintSDKError {
+    constructor(message: string, public readonly viemError: BaseError, code = WalletErrorCode.SEND_TRANSACTION_FAILED) {
+        super(message, code);
+    }
+}
+
+/**
+ * Error thrown when a transaction fails due to a contract execution error.
```

Comment:
Handling execution reverted issues tends to be the biggest pain point for devs in checkout and it may be here too

I would make instructions in this TSDoc as easy as possible

For example we could explain: when to use the viemError vs revertError, and show an example error object maybe?

Reply:
I added some field descriptions and an example of how to catch a revert error and interpret it